src/components/ProductMilestoneMarkModal/ProductMilestoneMarkModal.tsx

```diff
@@ -0,0 +1,70 @@
+import { Button, Modal } from '@patternfly/react-core';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { ProductMilestone, ProductVersion } from 'pnc-api-types-ts';
+
+import { useServiceContainer } from 'hooks/useServiceContainer';
+
+import * as productVersionApi from 'services/productVersionApi';
+
+import { createSafePatch } from 'utils/patchHelper';
+
+interface IProductMilestoneMarkModalProps {
+  productVersion: ProductVersion;
+  productMilestone: ProductMilestone;
+}
+export const ProductMilestoneMarkModal = ({ productVersion, productMilestone }: IProductMilestoneMarkModalProps) => {
+  const navigate = useNavigate();
+
+  const serviceContainerProductVersionPatch = useServiceContainer(productVersionApi.patchProductVersion);
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const handleModalToggle = () => {
+    setIsModalOpen((isModalOpen) => !isModalOpen);
+  };
+
+  const handleModalConfirm = () => {
+    const patchData = createSafePatch(productVersion, { currentProductMilestone: { id: productMilestone.id } });
+
+    serviceContainerProductVersionPatch
+      .run({ serviceData: { id: productVersion.id, patchData } })
+      .then(() => navigate('../milestones'))
+      .catch(() => {
+        throw new Error('Failed to edit current Product Milestone.');
+      });
+
+    handleModalToggle();
+  };
+
+  return (
+    <>
+      <Button
+        variant="tertiary"
+        isSmall
+        onClick={handleModalToggle}
+        isDisabled={productMilestone.id === productVersion?.currentProductMilestone?.id}
```

Comment:
- [done] user is not logged in, otherwise http request results in 500 (more over status code 500 is actually invalid, I reported new jira for it NCL-7888)
- [done]  milestone is already current
- [done] milestone is closed